### PR TITLE
fix : Some functions do not free heap memory when they exit.

### PR DESCRIPTION
**Pull request by fansehep**
fix : Some functions do not free heap memory when they exit.

### CODE REVIEW

File: src/benchInsert.c [hunk 1]
```diff
@@ -1144,6 +1155,7 @@ void *syncWriteProgressive(void *sarg) {
               pThreadInfo->end_table_to);
     int32_t *code = calloc(1, sizeof(int32_t));
     if (code == NULL) {
+        free(code);
```
Comment (sangshuduo) [inline src/benchInsert.c:1158]: we can't deref a NULL pointer.

File: src/benchInsert.c [hunk 2]
```diff
@@ -1368,6 +1380,7 @@ void *syncWriteProgressive(void *sarg) {
             pThreadInfo->cntDelay++;
             current_delay_node = calloc(1, sizeof(delayNode));
             if (current_delay_node == NULL) {
+                free(current_delay_node);
```
Comment (sangshuduo) [inline src/benchInsert.c:1383]: we can't free a NULL pointer

File: src/benchInsert.c (hunk 1 above)
Comment (fansehep) [reply src/benchInsert.c]: Sorry, I wasn't paying attention.

**Review by sangshuduo — CHANGES_REQUESTED** [review]
see my comments

**Review by zhaoyanggh — APPROVED** [review]
LGTM

**Comment by fansehep** [comment]
> LGTM
Thank you!

**Comment by sangshuduo** [comment]
Please follow the commitlint rule.

✖   subject may not be empty [subject-empty]
✖   type may not be empty [type-empty]

✖   found 2 problems, 0 warnings
ⓘ   Get help: https://github.com/conventional-changelog/commitlint/#what-is-commitlint

File: src/benchInsert.c [hunk 3]
```diff
@@ -638,7 +648,7 @@ static int startMultiThreadCreateChildTable(int db_index, int stb_index) {
 
     for (int64_t i = 0; i < threads; i++) {
         threadInfo *pThreadInfo = infos + i;
-        pThreadInfo->threadID = (int)i;
+        pThreadInfo->threadID = (uint32_t)i;
```
Comment (sangshuduo) [inline src/benchInsert.c:651]: converting with  (pthread_t) is better

**Review by sangshuduo — COMMENTED** [review]
converting suggestion

**Comment by fansehep** [comment]
> converting suggestion
ok

**Comment by sangshuduo** [comment]
@fansehep the commit log need to follow the commitlint rule too. Thanks.


--
Error: You have commit messages with errors

⧗   input: fix issues in benchInsert.c

fix issues in benchInsert.c

fix issues in benchInsert.c

fix issues in benchInsert.c
✖   subject may not be empty [subject-empty]
✖   type may not be empty [type-empty]

✖   found 2 problems, 0 warnings
ⓘ   Get help: https://github.com/conventional-changelog/commitlint/#what-is-commitlint

**Comment by fansehep** [comment]
> @fansehep the commit log need to follow the commitlint rule too. Thanks.
> 
> -- Error: You have commit messages with errors
> 
> ⧗ input: fix issues in benchInsert.c
> 
> fix issues in benchInsert.c
> 
> fix issues in benchInsert.c
> 
> fix issues in benchInsert.c heavy_multiplication_x subject may not be empty [subject-empty] heavy_multiplication_x type may not be empty [type-empty]
> 
> heavy_multiplication_x found 2 problems, 0 warnings ⓘ Get help: https://github.com/conventional-changelog/commitlint/#what-is-commitlint

so i should install the software and change my git log ? or just change it in my git commit log?

**Comment by sangshuduo** [comment]
> > @fansehep the commit log need to follow the commitlint rule too. Thanks.
> > -- Error: You have commit messages with errors
> > ⧗ input: fix issues in benchInsert.c
> > fix issues in benchInsert.c
> > fix issues in benchInsert.c
> > fix issues in benchInsert.c heavy_multiplication_x subject may not be empty [subject-empty] heavy_multiplication_x type may not be empty [type-empty]
> > heavy_multiplication_x found 2 problems, 0 warnings ⓘ Get help: https://github.com/conventional-changelog/commitlint/#what-is-commitlint
> 
> so i should install the software and change my git log ? or just change it in my git commit log?

commitlint is a Github actions script. it checks if the commit log follows the correct rule or not.
The easiest way might be to resubmit a PR with the correct commit log. Otherwise, you need to use git rebase to modify the commit log then resubmit PR.

**Comment by fansehep** [comment]
> > > @fansehep提交日志也需要遵循 commitlint 规则。谢谢。
> > > -- 错误：您的提交消息有错误
> > > ⧗ 输入：修复 benchInsert.c
> > > 中的问题 修复 benchInsert.c 中的问题
> > > 修复benchInsert.c 中的问题 修复 benchInsert.c 中的
> > > 问题可能不为空 [type-empty] 
> > > heavy_multiplication_x 发现 2 个问题，0 个警告 ⓘ 获取帮助：[https ://github.com/conventional-changelog/commitlint/#what-is-commitlint](https://github.com/conventional-changelog/commitlint/#what-is-commitlint)
> > 
> > 
> > 所以我应该安装软件并更改我的 git 日志？或者只是在我的 git 提交日志中更改它？
> 
> commitlint 是一个 Github 动作脚本。它检查提交日志是否遵循正确的规则。 最简单的方法可能是使用正确的提交日志重新提交 PR。否则，需要使用 git rebase 修改提交日志，然后重新提交 PR。

like this ?

**Comment by sangshuduo** [comment]
> > > > @fansehep提交日志也需要遵循 commitlint 规则。谢谢。
> > > > -- 错误：您的提交消息有错误
> > > > ⧗ 输入：修复 benchInsert.c
> > > > 中的问题 修复 benchInsert.c 中的问题
> > > > 修复benchInsert.c 中的问题 修复 benchInsert.c 中的
> > > > 问题可能不为空 [type-empty]
> > > > heavy_multiplication_x 发现 2 个问题，0 个警告 ⓘ 获取帮助：[https ://github.com/conventional-changelog/commitlint/#what-is-commitlint](https://github.com/conventional-changelog/commitlint/#what-is-commitlint)
> > > 
> > > 
> > > 所以我应该安装软件并更改我的 git 日志？或者只是在我的 git 提交日志中更改它？
> > 
> > 
> > commitlint 是一个 Github 动作脚本。它检查提交日志是否遵循正确的规则。 最简单的方法可能是使用正确的提交日志重新提交 PR。否则，需要使用 git rebase 修改提交日志，然后重新提交 PR。
> 
> like this ? 

Yes, you can refer to https://github.com/conventional-changelog/commitlint/#what-is-commitlint for details.

**Comment by sangshuduo** [comment]
> > > > @fansehep提交日志也需要遵循 commitlint 规则。谢谢。
> > > > -- 错误：您的提交消息有错误
> > > > ⧗ 输入：修复 benchInsert.c
> > > > 中的问题 修复 benchInsert.c 中的问题
> > > > 修复benchInsert.c 中的问题 修复 benchInsert.c 中的
> > > > 问题可能不为空 [type-empty]
> > > > heavy_multiplication_x 发现 2 个问题，0 个警告 ⓘ 获取帮助：[https ://github.com/conventional-changelog/commitlint/#what-is-commitlint](https://github.com/conventional-changelog/commitlint/#what-is-commitlint)
> > > 
> > > 
> > > 所以我应该安装软件并更改我的 git 日志？或者只是在我的 git 提交日志中更改它？
> > 
> > 
> > commitlint 是一个 Github 动作脚本。它检查提交日志是否遵循正确的规则。 最简单的方法可能是使用正确的提交日志重新提交 PR。否则，需要使用 git rebase 修改提交日志，然后重新提交 PR。
> 
> like this ? 

Yes, you can refer to https://github.com/conventional-changelog/commitlint/#what-is-commitlint for details.

**Comment by fansehep** [comment]
> > > > > @fansehep提交日志也需要遵循 commitlint 规则。谢谢。
> > > > > -- 错误：您的提交消息有错误
> > > > > ⧗ 输入：修复 benchInsert.c
> > > > > 中的问题 修复 benchInsert.c 中的问题
> > > > > 修复benchInsert.c 中的问题 修复 benchInsert.c 中的
> > > > > 问题可能不为空 [type-empty]
> > > > > heavy_multiplication_x 发现 2 个问题，0 个警告 ⓘ 获取帮助：[https ://github.com/conventional-changelog/commitlint/#what-is-commitlint](https://github.com/conventional-changelog/commitlint/#what-is-commitlint)
> > > > 
> > > > 
> > > > 所以我应该安装软件并更改我的 git 日志？或者只是在我的 git 提交日志中更改它？
> > > 
> > > 
> > > commitlint 是一个 Github 动作脚本。它检查提交日志是否遵循正确的规则。 最简单的方法可能是使用正确的提交日志重新提交 PR。否则，需要使用 git rebase 修改提交日志，然后重新提交 PR。
> > 
> > 
> > like this ? 
> 
> Yes, you can refer to https://github.com/conventional-changelog/commitlint/#what-is-commitlint for details.

Should i reopen a pull request？i have change the git log and push .

**Comment by sangshuduo** [comment]
> > > > > > @fansehep提交日志也需要遵循 commitlint 规则。谢谢。
> > > > > > -- 错误：您的提交消息有错误
> > > > > > ⧗ 输入：修复 benchInsert.c
> > > > > > 中的问题 修复 benchInsert.c 中的问题
> > > > > > 修复benchInsert.c 中的问题 修复 benchInsert.c 中的
> > > > > > 问题可能不为空 [type-empty]
> > > > > > heavy_multiplication_x 发现 2 个问题，0 个警告 ⓘ 获取帮助：[https ://github.com/conventional-changelog/commitlint/#what-is-commitlint](https://github.com/conventional-changelog/commitlint/#what-is-commitlint)
> > > > > 
> > > > > 
> > > > > 所以我应该安装软件并更改我的 git 日志？或者只是在我的 git 提交日志中更改它？
> > > > 
> > > > 
> > > > commitlint 是一个 Github 动作脚本。它检查提交日志是否遵循正确的规则。 最简单的方法可能是使用正确的提交日志重新提交 PR。否则，需要使用 git rebase 修改提交日志，然后重新提交 PR。
> > > 
> > > 
> > > like this ? 
> > 
> > 
> > Yes, you can refer to https://github.com/conventional-changelog/commitlint/#what-is-commitlint for details.
> 
> Should i reopen a pull request？i have change the git log and push .

Seems the script still detects the commit log is not good. Please resubmit the PR with a proper log. thanks.

**Pull request closed without merging**